ansible/roles/vault/tasks/init.yml

```diff
@@ -0,0 +1,61 @@
+---
+- name: Check if initialized
+  uri:
+    url: http://127.0.0.1:8200/v1/sys/init
+    method: GET
+  tags: init
+  register: init
+
+- name: Init vault if not initialized
+  uri:
+    url: http://127.0.0.1:8200/v1/sys/init
+    method: PUT
+    body: "{\"secret_shares\": 1, \"secret_threshold\": 1}"
+    body_format: json
+  when: not init.json.initialized
+  tags: init
+  register: key
+
+- name: Ensures key dir exists
+  file: 
+    path: "~/.vault/" 
+    state: directory
+  delegate_to: 127.0.0.1
+  tags: init
+  when: not init.json.initialized
+
+- name: Write key to file
+  copy:
+    content: "{{ key.json }}"
+    dest: "~/.vault/key.json"
+  delegate_to: 127.0.0.1
+  tags: init
+  when: not init.json.initialized
+
+- name: Check if sealed
+  uri:
+    url: http://127.0.0.1:8200/v1/sys/seal-status
+    method: GET
+  tags: init
+  register: sealed
+
+- name: Read in key
+  include_vars:
+    file: "~/.vault/key.json"
+    name: key
+  tags: init
+  when: sealed.json.sealed
+
+- name: Unseal
```

Comment:
slight duplication of unseal

Reply:
Fair. But I'm not sure if moving this into a different file to include would actually make the code more readable. I don't expect  to repeat this a third time